File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    zip_board = list(map(list, zip(*board)))
    stack, answer = [], 0
    for m in moves:
        i = m - 1
        for j in range(len(board)):
            if zip_board[i][j] != 0:
                if stack and stack[-1] == zip_board[i][j]:
                    stack.pop()
                    answer += 2
                else:
                    stack.append(zip_board[i][j])
                zip_board[i][j] = 0
                break
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        Stack<Integer> stack = new Stack<>();
        int answer = 0;
        for (int move : moves) {
            int j = move - 1;
            for (int i = 0; i < board.length; i++) {
                if (board[i][j] != 0) {
                    if (!stack.isEmpty() && stack.peek() == board[i][j]) {
                        stack.pop();
                        answer += 2;
                    } else {
                        stack.push(board[i][j]);
                    }
                    board[i][j] = 0;
                    break;
                }
            }
        }
        return answer;
    }
}